src/views/self/view_weapons.cpp

```diff
@@ -176,5 +176,104 @@ namespace big
 			ImGui::SliderFloat("Distance", &g.weapons.aimbot.distance, 1.f, 1000.f, "%.0f");
 			ImGui::PopItemWidth();
 		}
+
+		ImGui::SeparatorText("Ammunation");
+		static Hash selected_weapon_hash, selected_weapon_attachment_hash{};
+		static std::string selected_weapon, selected_weapon_attachment;
+		ImGui::PushItemWidth(300);
+		if (ImGui::BeginCombo("Weapons", selected_weapon.c_str()))
+		{
+			for (auto& weapon : g_gta_data_service->weapons())
+			{
+				bool is_selected = weapon.second.m_hash == selected_weapon_hash;
+				if (weapon.second.m_display_name != "NULL" && ImGui::Selectable(weapon.second.m_display_name.c_str(), is_selected, ImGuiSelectableFlags_None))
+				{
+					selected_weapon      = weapon.second.m_display_name;
+					selected_weapon_hash = weapon.second.m_hash;
+				}
+				if (is_selected)
+					ImGui::SetItemDefaultFocus();
+			}
+			ImGui::EndCombo();
+		}
+		ImGui::PopItemWidth();
+		ImGui::SameLine();
+		if (ImGui::Button("Give Weapon"))
```

Comment:
All of these ImGui::Button with queue_job inside the if can be replaced with components::button(text, callback) calls

Reply:
Thanks. I just straight copied them from my fork from BBv2, didn't know you had wrappers.